### PR TITLE
show feedback when starting a new connection

### LINKED ISSUE #807: showing feedback when starting a new connection
we're currently not displaying any feedback when creating a new connection, but we should display something like "Connected to {alias}". This should be configurable and turned off when `feedback=0`

Likewise, the "switching to connection" messages should be turned off when `feedback=0`

you can get the current feedback level with: https://github.com/ploomber/jupysql/blob/master/src/sql/_current.py

### PR DESCRIPTION
**Pull request by bbeat2782**
## Describe your changes

Show feedback when starting a new connection. This is configurable through setting `%config SqlMagic.feedback = 0` 

## Issue number

Closes #807 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--834.org.readthedocs.build/en/834/

### CODE REVIEW

File: src/tests/test_connection.py [hunk 1]
```diff
@@ -717,12 +717,18 @@ def test_feedback_when_switching_connection_with_existing_connection(
     assert "Switching to connection one" == captured.out.splitlines()[-1]
 
 
-def test_no_feedback(ip_empty, tmp_empty, capsys):
+@pytest.mark.parametrize(
```
Comment (edublancas) [inline src/tests/test_connection.py:720]: this needs a bit more testing, can you check that the message is shown both when feedback is 1,2 and that it isn't shown when feedback is 0?

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/tests/test_connection.py (hunk 1 above)
Comment (bbeat2782) [reply src/tests/test_connection.py:720]: There was testing feedbacks when feedback=0, so I changed the name of the function: [fixed](https://github.com/ploomber/jupysql/pull/834/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R740-R749)

For feedback=1 and 2, [fixed](https://github.com/ploomber/jupysql/pull/834/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R723-R726)

File: src/sql/connection/connection.py [hunk 2]
```diff
@@ -270,15 +270,14 @@ def set(
...
                         identifier = alias or descriptor
-                        display.message(
-                            f"Connecting and switching to connection {identifier}"
-                        )
+                        if cls.current and cls.current.alias != alias:
+                            display.message(
+                                f"Connecting and switching to connection {identifier}"
+                            )
+                        else:
+                            display.message(f"Connecting to connection {identifier}")
```
Comment (edublancas) [inline src/sql/connection/connection.py:280]:
```suggestion
                            display.message(f"Connecting to {identifier!r}")
```

File: src/sql/connection/connection.py [hunk 3]
```diff
@@ -270,15 +270,14 @@ def set(
...
-                        and cls.current.alias != alias
-                    ):
+                    if _current._config_feedback_normal_or_more():
                         identifier = alias or descriptor
-                        display.message(
-                            f"Connecting and switching to connection {identifier}"
-                        )
+                        if cls.current and cls.current.alias != alias:
+                            display.message(
+                                f"Connecting and switching to connection {identifier}"
```
Comment (edublancas) [inline src/sql/connection/connection.py:277]:
```suggestion
                                f"Connecting and switching to connection {identifier!r}"
```

**Review by edublancas — CHANGES_REQUESTED** [review]
suggested some changes in the messages.

also found a problem: when connecting to a db that requires a password, the message prints the raw URL (including the password). switching to the connection doesn't have this problem. please fix and add a unit test

example: 
[connecting.ipynb.zip](https://github.com/ploomber/jupysql/files/12470391/connecting.ipynb.zip)


you can mock `sqlalchemy.create_engine` so it doesn't throw an error when trying to connect to the database (since there is no database running in the background), we have some tests where we're doing this, ping me if you need help

**Comment by bbeat2782** [comment]
@edublancas 
I'm a little bit confused about this part.
> when connecting to a db that requires a password, the message prints the raw URL (including the password). switching to the connection doesn't have this problem.

Are you saying that the password should be replaced with ***? 

But from the notebook you provided, I found a message with `Connecting and switching to connection postgresql://ploomber_app:ploomber_app_password@localhost:5432/db`, which is already in the master branch. So should I change it for the connecting message and connecting and switching message?

**Comment by edublancas** [comment]
> Are you saying that the password should be replaced with ***?

Yes, we should not print the password since this is sensitive information. In the connection object, I think we already have a version of the URL with the password hidden (replaced with ***) so there's no need for you to write that logic, it should be there already

> But from the notebook you provided, I found a message with Connecting and switching to connection postgresql://ploomber_app:ploomber_app_password@localhost:5432/db, which is already implemented in the master branch. So should I change it for the connecting message and connecting and switching message?

I'm not following this part.

**Comment by bbeat2782** [comment]
> I'm not following this part.

Referring to this part of the notebook. In the last cell, the password isn't replaced with ***.

Screen Shot 2023-08-30 at 3 07 04 PM

**Comment by edublancas** [comment]
> Referring to this part of the notebook. In the last cell, the password isn't replaced with ***. So I think password in such case should also be replaced with ***.

right, it should also be replaced with ***

File: src/sql/connection/connection.py (hunk 2 above)
Comment (bbeat2782) [reply src/sql/connection/connection.py]: [fixed](https://github.com/ploomber/jupysql/pull/834/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R293)

File: src/sql/connection/connection.py (hunk 3 above)
Comment (bbeat2782) [reply src/sql/connection/connection.py]: [fixed](https://github.com/ploomber/jupysql/pull/834/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R290)

**Comment by bbeat2782** [comment]
@edublancas 
Because the url with password replaced with *** is created after  `from_connect_str`, I relocated the part that displays the message: [fixed](https://github.com/ploomber/jupysql/pull/834/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R286-R293)

test functions for password replaced with *** : [here](https://github.com/ploomber/jupysql/pull/834/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R752-R779)

**Review by edublancas — APPROVED** [review]

**Pull request merged**